awswrangler/_utils.py

```diff
@@ -401,3 +402,33 @@ def check_schema_changes(columns_types: Dict[str, str], table_input: Optional[Di
                     f"Schema change detected: Data type change on column {c} "
                     f"(Old type: {catalog_cols[c]} / New type {t})."
                 )
+
+
+def pylist_to_arrow(
```

Comment:
This looks like it should be two separate functions (split into two if statements and two returns) rather than one parametrised by type

Reply:
I took that function directly from [Arrow](https://github.com/apache/arrow/blob/master/python/pyarrow/table.pxi#L5134) as it's only available in pyarrow 7+ and we support below. That being said, I only need to handle the Table return so can remove the other one